lib/trustly/data/base.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Trustly
+  module Data
+    class Base
+      attr_accessor :payload
+
+      def initialize(**_options)
+        self.payload = {}
+      end
+
+      def to_json(*_args)
+        payload.to_json
+      end
+
+      private
+
+      # Vacuum out all keys being set to nil in the data to be communicated
+      def vacuum(data)
+        case data
+        when Array then vacuum_array_data(data)
+        when Hash then vacuum_hash_data(data)
+        else data
+        end
+      end
+
+      def vacuum_array_data(data)
+        ret = data.each_with_object([]) do |element, acc|
+          processed_element = vacuum(element) unless element.nil?
+          next if processed_element.nil?
+
+          acc.push(processed_element)
+        end
+        ret.length.zero? ? nil : ret
```

Comment:
```suggestion
        ret.empty? ? nil : ret
```

Or `ret.presence` if you're willing to add a dependency to `active_support/core_ext`.